src/buildstream/_loader/loadelement.pyx

```diff
@@ -379,44 +377,11 @@ cdef class LoadElement:
         self._dep_cache = FrozenBitMap(self._dep_cache)
 
 
-def _dependency_cmp(Dependency dep_a, Dependency dep_b):
-    cdef LoadElement element_a = dep_a.element
-    cdef LoadElement element_b = dep_b.element
-
-    # Sort on inter element dependency first
-    if element_a.depends(element_b):
-        return 1
-    elif element_b.depends(element_a):
-        return -1
-
-    # If there are no inter element dependencies, place
-    # runtime only dependencies last
-    if dep_a.dep_type != dep_b.dep_type:
-        if dep_a.dep_type == DependencyType.RUNTIME:
-            return 1
-        elif dep_b.dep_type == DependencyType.RUNTIME:
-            return -1
-
-    # All things being equal, string comparison.
-    if element_a.name > element_b.name:
-        return 1
-    elif element_a.name < element_b.name:
-        return -1
-
-    # Sort local elements before junction elements
-    # and use string comparison between junction elements
-    if element_a.junction and element_b.junction:
-        if element_a.junction > element_b.junction:
-            return 1
-        elif element_a.junction < element_b.junction:
-            return -1
-    elif element_a.junction:
-        return -1
-    elif element_b.junction:
-        return 1
-
-    # This wont ever happen
-    return 0
+def _cmp_key(Dependency dep):
+    if element.junction:
```

Comment:
If it's desired to use DependencyType, we could make first item in tuple an integer with value 0, 1, 2 corresponding build vs runtime vs all types.

Reply:
Oh. Now I get why it's done like how it was.